### PR TITLE
Moved core documentation to terra dev site.

### LINKED ISSUE #158: Moving Documentation: Core
# Story

## Description

As an engineer, I would like to move over documentation to the new dev-site.

## Motivation

So that the new dev-site is a more complete resource for consumers.

## Definition of Done

We'll know we're done when the following files are moved over and updated where necessary:

- [Graph](https://github.com/cerner/terra-graphs/blob/7b84660ad711ec66d03fe3d0e7eaca0756735a54/packages/terra-graphs-docs/docs/core/Graph.md)
- [Lifecycle](https://github.com/cerner/terra-graphs/blob/7b84660ad711ec66d03fe3d0e7eaca0756735a54/packages/terra-graphs-docs/docs/core/Lifecycle.md)
- [Axes](https://github.com/cerner/terra-graphs/blob/7b84660ad711ec66d03fe3d0e7eaca0756735a54/packages/terra-graphs-docs/docs/helpers/Axes.md)
- [Dateline](https://github.com/cerner/terra-graphs/blob/7b84660ad711ec66d03fe3d0e7eaca0756735a54/packages/terra-graphs-docs/docs/helpers/Dateline.md)
- [Eventline](https://github.com/cerner/terra-graphs/blob/7b84660ad711ec66d03fe3d0e7eaca0756735a54/packages/terra-graphs-docs/docs/helpers/Eventline.md)
- [Ticks](https://github.com/cerner/terra-graphs/blob/7b84660ad711ec66d03fe3d0e7eaca0756735a54/packages/terra-graphs-docs/docs/helpers/Ticks.md)

### PR DESCRIPTION
**Pull request by AshishMotanamGurunadham**
### Summary

- Moved core documentation(lifecycle, graph, axes, eventline, dateline, ticks) to terra-dev-site.

Closes #158 

### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

**Review by tbiethman — APPROVED** [review]

**Comment by mjhenkes** [comment]
Could we get all doc nested under the 'graphs' folder? once this gets merged into terra dev site, the core folder is going to get lost in amongst all the other components.

File: packages/terra-graphs-docs/src/terra-dev-site/doc/Core/Lifecycle.2.doc.mdx [hunk 1]
```diff
@@ -104,4 +106,4 @@ Separation of _container_ and _content_ allows us to:
 -   Draw different combinations of graphs i.e. we can draw a _Combination graph_ with _2 Bar graphs_ or _1 Bar graph_ and _1 line graph_ without ever needing to building those natively in Carbon.
 -   Draw different combinations of graph in any order necessary.
 
-![Alt](../assets/content-interface.png "GraphContent Interface")
+<img src= { graphContent } alt="Graph Content" />
```
Comment (mjhenkes) [inline packages/terra-graphs-docs/src/terra-dev-site/doc/Core/Lifecycle.2.doc.mdx:109]: You might want to scale this image down, it's a bit large.

Comment (AshishMotanamGurunadham) [reply packages/terra-graphs-docs/src/terra-dev-site/doc/Core/Lifecycle.2.doc.mdx]: Adjusted as needed [here](https://github.com/cerner/terra-graphs/pull/179/commits/fd0857beac0da90898defb7ab72d686d952409e9#diff-bf4593511b860bc910146c3c868292ec3f99a140967ced3f092b3f0e484b2da6R109).

**Comment by AshishMotanamGurunadham** [comment]
> Could we get all doc nested under the 'graphs' folder? once this gets merged into terra dev site, the core folder is going to get lost in amongst all the other components.

Moved [core documentation](https://github.com/cerner/terra-graphs/pull/179/commits/fd0857beac0da90898defb7ab72d686d952409e9#) to graphs folder.

**Review by mjhenkes — APPROVED** [review]
I think this is ok for now but we should take the time to review the docs again before we publish them to terra-ui.

**Review by sdadn — APPROVED** [review]

**Pull request merged**